source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -742,6 +747,66 @@ The event API here is assumed to be like the standard `Python Event
           interrupt connection read
           close tempConnection
 
+RTT thread
+``````````
+
+The requirements in the `Measuring RTT`_ section can be satisfied with an
+addtional thread that periodically runs the isMaster command on a dedicated
+connection, for example:
+
+.. code-block:: python
+
+  class RttMonitor(Thread):
+    def __init__():
+        # Options:
+        serverAddress = serverAddress
+        connectTimeoutMS = connectTimeoutMS
+        heartbeatFrequencyMS = heartbeatFrequencyMS
+        # Internal state:
+        connection = Null
+        lock = Mutex()
+        movingAverage = MovingAverage()
+
+    def reset():
+        with lock:
+            movingAverage.reset()
+
+    def addSample(rtt):
+        with lock:
+            movingAverage.update(rtt)
+
+    def average():
+        with lock:
+            return movingAverage.get()
+
+    def run():
+        while this monitor is not stopped:
+            try:
+                rtt = pingServer()
+                addSample(rtt)
+            except (NetworkError, CommandError) as exc:
```

Comment:
as I understand any other exceptions except `NetworkError, CommandError` is not possible here. if so, will it be more accurate to have only `except (Exception)` line here?

Reply:
Good point, I updated to catch all exceptions.